easy-ui-react/src/Button/Button.module.scss

```diff
@@ -0,0 +1,285 @@
+@import "../styles/common";
+
+@mixin link-button($color, $hover-color, $active-color, $box-shadow) {
+  color: $color;
+  background-color: transparent;
+  border: var(--ezui-shape-border-width-1) solid transparent;
+  text-decoration: underline;
+
+  &:focus {
+    border: var(--ezui-shape-border-width-1) solid $color;
+    box-shadow: $box-shadow;
+    outline: none;
+  }
+
+  &:disabled {
+    color: var(--ezui-color-gray-300);
+    pointer-events: none;
+  }
+
+  &:active:not(:disabled) {
+    color: $active-color;
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    color: $hover-color;
+  }
+}
+
+@mixin filled-button(
+  $color,
+  $active-color,
+  $background,
+  $hover-focus-background,
+  $active-background,
+  $box-shadow
+) {
+  background-color: $background;
+  color: $color;
+
+  &:focus {
+    background-color: $hover-focus-background;
+    color: $color;
+    box-shadow: $box-shadow;
+    outline: none;
+  }
+
+  &:disabled {
+    color: var(--ezui-color-white);
+    background-color: var(--ezui-color-gray-200);
+    pointer-events: none;
+  }
+
+  &:link {
+    text-decoration: none;
+  }
+
+  &:not(:disabled):active {
+    color: $active-color;
+    background-color: $active-background;
+    &:focus {
+      box-shadow: none;
+    }
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    background-color: $hover-focus-background;
+    color: $color;
+  }
+}
+
+@mixin outlined-button($border, $hover-border, $active-border, $box-shadow) {
+  border: var(--ezui-shape-border-width-1) solid $border;
+  color: $border;
+  background-color: transparent;
+
+  &:focus {
+    border: var(--ezui-shape-border-width-1) solid $border;
+    box-shadow: $box-shadow;
+    outline: none;
+  }
+
+  &:disabled {
+    color: var(--ezui-color-gray-300);
+    border: var(--ezui-shape-border-width-1) solid var(--ezui-color-gray-300);
+    pointer-events: none;
+  }
+
+  &:link {
+    text-decoration: none;
+  }
+
+  &:active:not(:disabled) {
+    color: $active-border;
+    border: var(--ezui-shape-border-width-1) solid $active-border;
+    &:focus {
+      box-shadow: none;
+    }
+  }
+
+  &:hover:not(:disabled):not(:active) {
+    color: $hover-border;
+    border: var(--ezui-shape-border-width-1) solid $hover-border;
+  }
+}
+
+.Button {
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: var(--ezui-shape-border-radius-base);
+  border: none;
+
+  &.colorPrimaryFilled {
+    @include filled-button(
+      var(--ezui-color-white),
+      var(--ezui-color-white),
+      var(--ezui-color-blue-500),
+      var(--ezui-color-blue-600),
+      var(--ezui-color-blue-700),
+      var(--ezui-shadow-button)
+    );
+  }
```

Comment:
in CSS modules i don't believe we need to nest it due to how modules isolates it already to the component. could help slightly IMO with cognitive overhead